bootstrap.js

```diff
@@ -0,0 +1,46 @@
+/* global Components */
+
+const { classes: Cc, interfaces: Ci } = Components;
+const DefaultUA = Cc["@mozilla.org/network/protocol;1?name=http"]
+  .getService(Ci.nsIHttpProtocolHandler).userAgent;
+const ObserverService = Cc["@mozilla.org/observer-service;1"].getService(Ci.nsIObserverService);
+
+class UAOverrider {
+  init() {
+    ObserverService.addObserver(this, "http-on-useragent-request", false);
```

Comment:
Suggest to use NS_HTTP_ON_USERAGENT_REQUEST_TOPIC instead of "http-on-useragent-request" directly?